crates/bevy_ecs/src/reflect/bundle.rs

```diff
@@ -85,8 +86,19 @@ impl ReflectBundle {
     }
 
     /// Removes this [`Bundle`] type from the entity. Does nothing if it doesn't exist.
-    pub fn remove(&self, entity: &mut EntityWorldMut) {
-        (self.0.remove)(entity);
+    pub fn remove(&self, entity: &mut EntityWorldMut) -> ReflectBundle {
+        (self.0.remove)(entity)
+    }
+
+
+    /// Removes all components in the [`Bundle`] from the entity and returns their previous values.
+    ///
+    /// **Note:** If the entity does not have every component in the bundle, this method will not
+    /// remove any of them.
+    // TODO: BundleRemover?
```

Comment:
This comment is very unclear to me :) I know it's draft but I want to make sure we don't forget.

Reply:
It's copy pasted from ecs's take for now